app/controllers/products_controller.rb

```diff
@@ -1,5 +1,54 @@
 # frozen_string_literal: true
 
 class ProductsController < ApplicationController
-  def index; end
+  def index
+    @products = Product.all
+  end
+
+  def show
+    @product = Product.find(params[:id])
```

Comment:
let's move this finder to private method `get_product` which will be called for show, edit, update and destroy methods, use `before_filter` for this